src/styles/blocks/search-bar.css

```diff
@@ -0,0 +1,68 @@
+.search-form {
+  margin-top: 20px;
+}
+
+.search-form__label {
+  display: flex;
+  align-items: center;
+  box-sizing: border-box;
+
+  border-radius: 3px;
+  border: 1px solid #e1e7ed;
+  box-shadow: 0 1px 8px 0 #3d4e611a;
+}
+
+.search-form__input {
+  color: inherit;
+  font-size: inherit;
+  font-weight: inherit;
+  font-family: inherit;
+  line-height: inherit;
+
+  margin: 0;
+  padding: 0;
+  flex-grow: 1;
+  border: none;
+  outline: none;
+}
+
+.search-form__input::placeholder {
+  color: inherit;
+  font-size: inherit;
+  font-weight: inherit;
+  font-family: inherit;
+  line-height: inherit;
+}
+
+.search-form__label:hover {
+  box-shadow: 0 4px 4px 0 #00000040;
+}
+
+.search-form__label:has(.search-form__input:focus) {
+  box-shadow: 0 4px 4px 0 #00000040;
+  background-color: linear-gradient(180deg, #fff 0%, #f6f6f7 100%);
+}
```

Comment:
The `:has()` pseudo-class is not supported in all browsers, which might cause issues with the intended styling. Consider using JavaScript to add a class when the input is focused, or ensure that the browsers you are targeting support this feature.